test: Add retries to increase reliability of test

danieljbruce opened the pull request:
This PR adds retries to the create table function in a test in order to reduce the risk of a flakey test in the case where the table cannot be created. This will solve some of the flakey test issues that we have been having.

Fixes #1292 
Fixes #1291

danieljbruce committed:
Skip a flakey test

This test needs to be skipped as it is causing P1 issues to appear. We need a formal plan for reproducing this issue.

danieljbruce committed:
Remove block of code for skipped test

This block of code is an after hook that will not work anymore now that the test is skipped because the test creates an instance and the after hook tries to delete it

danieljbruce committed:
Merge branch 'main' into skip-the-close-client-test

kolea2 commented on system-test/read-rows.ts:
```diff
@@ -85,7 +85,7 @@ describe('Bigtable/Table', () => {
 
   describe('close', () => {
-    it('should fail when invoking readRows with closed client', async () => {
+    it.skip('should fail when invoking readRows with closed client', async () => {
```
can we add retries to this test instead? I would prefer not to skip it, unless it is consistently failing.

danieljbruce replied:
Yes. I added a change which will still allow the test to run and retry in the appropriate circumstances.

danieljbruce committed:
Add retry options

Add retry options to create table to ensure that there is enough time for the instance to get created in order to avoid flakey tests

danieljbruce committed:
Merge branch 'main' into skip-the-close-client-test

danieljbruce committed:
Merge branch 'main' into skip-the-close-client-test
